libs/ngxtension/svg-sprite/src/svg-sprite.ts

```diff
@@ -0,0 +1,471 @@
+
+	/**
+	 * @ignore
+	 */
+	public ngOnInit() {
+		runInInjectionContext(this.injector, () => {
+			// Copy the 'viewBox' from the 'symbol' element in the sprite to this svg.
+			// Turn this effect in a 'noop' when the svg already has a 'viewBox'.
+			if (!this.element.hasAttribute('viewBox'))
+				effect(() => {
+					const element = this.element;
+					const autoViewBox = this.autoViewBox$();
+					const svg = this.svg$();
+					const fragment = this.fragment$();
+
+					if (!autoViewBox || svg == null || fragment == null) return;
+
+					try {
+						const viewBox = svg
+							.querySelector(`#${fragment}`)
+							?.getAttribute('viewBox');
+
+						if (viewBox == null) return;
+
+						element.setAttribute('viewBox', viewBox);
+					} catch {
+						// the querySelector could throw due to an invalid selector
+					}
+				});
+
+			// Create a 'use' element which instantiates a 'symbol' element of the sprite.
+			effect((beforeEach) => {
+				const fragment = this.fragment$();
+				const sprite = this.sprite$();
+				const spriteConfig = this.spriteConfig$();
+				const element = this.element;
+
+				let classes: string[] = [];
+
+				// Clear child nodes and remove old classes of this svg.
+				beforeEach(() => {
+					element.replaceChildren();
+					element.classList.remove(...classes);
+				});
+
+				if (fragment == null || sprite == null || spriteConfig == null) return;
+
+				const useElement = document.createElementNS(
+					element.namespaceURI,
+					'use',
+				);
+
+				// Add classes when provided.
+				if (spriteConfig.classes != null) {
+					const _classes = spriteConfig.classes(fragment);
+					classes =
+						typeof _classes === 'string'
+							? _classes.split(' ').filter(Boolean)
+							: _classes;
+					element.classList.add(...classes);
+				}
+
+				useElement.setAttribute(
+					'href',
+					spriteConfig.url(spriteConfig.baseUrl, fragment),
+				);
+
+				// Support old browsers. Modern browser will ignore this if they support 'href'.
+				useElement.setAttribute(
+					'xlink:href',
+					spriteConfig.url(spriteConfig.baseUrl, fragment),
+				);
+
+				element.appendChild(useElement);
+			});
+		});
+	}
+
+	/**
+	 * The `fragment` which identifies a `symbol` in this {@link NgxSvgSpriteFragment.sprite svg sprite}.
+	 */
+	public readonly fragment$ = signal<string | undefined>(undefined);
+
+	/**
+	 * The `name` of the {@link NgxSvgSprite svg sprite} this {@link NgxSvgSpriteFragment.fragment fragment} is a part of.
+	 *
+	 * @see {@link NgxSvgSprite.name}
+	 */
+	public readonly sprite$ = signal<string | undefined>(undefined);
+
+	/**
+	 * Whether `autoViewBox` is disabled.
+	 *
+	 * @see overrides {@link NgxSvgSprite.autoViewBox}
+	 */
+	public readonly autoViewBoxDisabled$ = signal<boolean>(false);
+
+	/**
+	 * @ignore
+	 */
+	private readonly spriteConfig$ = computed(() => {
+		const sprite = this.sprite$();
+
+		if (sprite != null) return this.sprites.get(sprite);
+
+		return undefined;
```

Comment:
suggestion: since `this.sprites.get()` returns the sprite from a Record, we won't get a runtime error and will just get `undefined` if the sprite isn't there. Feel free to skip this suggestion though if you think explicitly checking for null is better.

```suggestion
		return this.sprites.get(this.sprite$());
```

Reply:
I like the check better, because `null` and `undefined` are coerced to strings. A sprite _can_ be registered with the name `"null"`or `"undefined"`.